### ISSUE TITLE
Wardens confused after chunk loading after the chunk system rewrite

### ISSUE THREAD

**Issue by petersv5** [PaperSpigot/Paper#8481]
### Expected behavior

Wardens occupied by a repeating note block on chunk unload will still be occupied (aggrod) after loading the chunk again, or at least quickly reaquire the note block as a target.

### Observed/Actual behavior

Wardens lose all interest in the old target and fail to require the note block target when reloaded. The wardens eventually despawn due to a lack of aggro.

### Steps/models to reproduce

Any warden-based mob switch will show the effect.
Have a warden distracted by a repeating note block (4gt clock). Unload chunk by moving away or restarting the server. Return to the chunk and observe the warden not being interested in the repeating note block and eventually despawning.

### Plugin and Datapack List

None

### Paper version

This server is running Paper version git-Paper-"236a296" (MC: 1.19.2) (Implementing API version 1.19.2-R0.1-SNAPSHOT) (Git: 236a296)

Error was git bidect:ed to 01a13871deefa50e186a10b63f71c5e0459e7d30, the "Rewrite chunk system" committed 2022-09-26.

### Other

Error was git bidect:ed to 01a13871deefa50e186a10b63f71c5e0459e7d30, the "Rewrite chunk system" committed 2022-09-26.

**Comment by petersv5** [comment]
I have instrumented the code a bit. It seems that warden::shouldListen() is never called. 
A bit further seems to point to there not being a VibrationListener for the warden, or at least not one that has VibrationListener::handleGameEvent() being called. 

I am not sure how to dig further into this. I have not really ofund/understood code to create the VibrationListner instance when restoring a chunk.

**Comment by Owen1212055** [comment]
The issue is that on ``ServerLevel$EntityCallbacks#onTrackingStart`` ``entity.updateDynamicGameEventListener(DynamicGameEventListener::add);`` is called, which is supposed to register the dynamic listener into the chunk that the entity is loaded in. 

However, it seems that for whatever reason at this stage the chunk the entity is in isn't loaded yet. 
So, when ``DynamicGameEventListener#move`` is called this bit that is supposed to register the current listener into the
chunk returns null and doesn't do anything.
```java
ifChunkExists(world, this.lastSection, (dispatcher) -> {
                    dispatcher.register(this.listener);
                });
```

**Comment by petersv5** [comment]
It seems that the updateDynamicGameEventListener() gets called when then chunk is updating to BORDER. Is the issue that the update to BORDER is fired prematurely, before the chunk really has entered that state?

**Comment by petersv5** [comment]
It seems that in NewChunkHandler#handleFullStatusChange() the issue is created by
          this.changeEntityChunkStatus(ChunkHolder.FullChunkStatus.BORDER);
which leads to the DynamicGameEventListener#move trying to reference the chunk is called before 
          chunk.onChunkLoad(this);
which leads to ServerChunkCache#addLoadedChunk() being called.

If I reverse these two lines everything seems to work. I do not know if this new order:
          chunk.onChunkLoad(this);
          this.changeEntityChunkStatus(ChunkHolder.FullChunkStatus.BORDER);
is safe. Anyone more knowledgable?

**Comment by petersv5** [comment]
I created a PR #8483 with the change above.

**Comment by Spottedleaf** [comment]
Should be fixed by https://github.com/PaperMC/Paper/commit/f7da209586b12d169719e38b3dfa2664b6474a67

**Comment by petersv5** [comment]
A quick test shows that the problem is fixed in HEAD after the patch.
Thanks, and sorry for my failed patch attempt.

**Comment by lynxplay** [comment]
Resolved by f7da209586b12d169719e38b3dfa2664b6474a67
